### PR TITLE
Update colors in theme to new UI kit and update color story

### PR DESCRIPTION
**Pull request by gustaveen**
## Describe your changes
Update primitive colors in theme according to updated Figma https://www.figma.com/design/5kmmDdh6StpXzbEfr7WevV/Hedvig-UI-Kit?m=dev&node-id=15506-2412

Screenshot 2024-06-20 at 14 23 41

## Justify why they are needed
New UI kit
## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by gustaveen** [comment]
* **#4393** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4393?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @gustaveen and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by alebedev — APPROVED** [review]
Looks good

File: packages/ui/src/theme/colors/colors.stories.tsx [hunk 1]
```diff
@@ -29,31 +37,30 @@ type Props = {
 
 const Palette = ({ colors, name, prefix, suffix }: Props) => (
   <>
-    <ColorName>{name}</ColorName>
-    <ColorGrid>
+    <Heading className={sprinkles({ mb: 'md' })}>{name}</Heading>
```
Comment (alebedev) [inline packages/ui/src/theme/colors/colors.stories.tsx:40]: Not sure I like these shorthands, translating into `marginBottom` in my head hurts a little. This is unrelated to changes in this PR, of course

**Review by guilhermespopolin — APPROVED** [review]
